src/methods/call.js

```diff
@@ -332,15 +333,45 @@ export function OrdinaryCallEvaluateBody(
       try {
         F.isSelfRecursive = false;
         let effects = realm.evaluateForEffects(guardedCall);
+        let c = effects[0];
+        let v = c instanceof ReturnCompletion ? c.value : realm.intrinsics.undefined;
         if (F.isSelfRecursive) {
-          AbstractValue.reportIntrospectionError(F, "call to function that calls itself");
-          throw new FatalError();
           //todo: need to emit a specialized function that temporally captures the heap state at this point
+          let fullArgs = [F].concat(argumentsList);
```

Comment:
Just calling F here is not safe since it may be referring to outer scope variables and properties that will not be in the right state at the temporal point where this call happens.